### PR TITLE
Add support for unsafe requests in the integration DSL

### PR DESCRIPTION
**Pull request by KtorZ**
<p align="right">#173</p>

# Overview

- [x] I have added an `unsafeRequest` combinator that can be used to submit raw requests to the API, specifying methods, endpoint and body by hand (can't yet customize headers, but that's an improvement we could imagine)

- [x] I have added an extra `(</>)` combinator to easily create paths containing variables from ids

- [x] I have added an extra `expectJSONError` combinator to expect a response to return a JSON validation error containing a given message.

- [x] I have added a dummy integration test example that illustrate all the above.

# Comments

We may want to remove the example in the future as it serves no real purpose :)

### CODE REVIEW

**Review by piotr-iohk — APPROVED** [review]
nice! :+1:

File: test/integration/Test/Integration/Framework/Request.hs [hunk 1]
```diff
@@ -59,6 +80,47 @@ instance Request () where
...
+        , HasManager ctx
+        )
+    => (Method, Text)
+    -> Maybe Aeson.Value
+    -> m (Either ClientError a)
+unsafeRequest (verb, path) body = do
+    (base, man) <- view manager
+    req <- parseRequest (showBaseUrl $ base { baseUrlPath = T.unpack path })
+    res <- liftIO $ httpLbs (prepare req) man
+    return $ case responseStatus res of
```
Comment (akegalj) [inline test/integration/Test/Integration/Framework/Request.hs:98]: minor - but this looks like a good candidate for `if` (not `case`) as you used guards to be able to compare status

```haskell
let resStatus = responseStatus res
in return $
  if resStatus >= status200 && resStatus <= status300
     then...
     else ...
```

File: test/integration/Test/Integration/Scenario/Wallets.hs [hunk 2]
```diff
@@ -4,12 +4,24 @@ module Test.Integration.Scenario.Wallets
...
 
-
 spec :: Scenarios Context
 spec = do
+    scenario "demo unsafeRequest" $ do
+        fixture <- setup $ defaultSetup
+
+        let endpoint = "api" </> "v1" </> "wallets" </> fixture ^. wallet . walletId
+        response <- unsafeRequest ("PUT", endpoint) $ Just $ [json|{
+            "invalidField": #{fixture ^. wallet}
```
Comment (akegalj) [inline test/integration/Test/Integration/Scenario/Wallets.hs:18]: useful :+1:

**Review by akegalj — APPROVED** [review]

File: test/integration/Test/Integration/Framework/Request.hs (hunk 1 above)
Comment (KtorZ) [reply test/integration/Test/Integration/Framework/Request.hs:98]: Indeed. I was expecting more cases initially :sweat_smile:

**Pull request merged**